src/app/child-dev-project/schools/activities-overview/activities-overview.component.ts

```diff
@@ -24,30 +24,31 @@ export class ActivitiesOverviewComponent implements OnInitDynamicComponent {
   entity: Entity;
   records: RecurringActivity[] = [];
   listConfig: EntityListConfig;
-  activityConstructor = RecurringActivity;
 
   constructor(private entityMapper: EntityMapperService) {}
 
-  async onInitFromDynamicConfig(config: any) {
+  onInitFromDynamicConfig(config: any) {
     if (config?.config?.columns) {
       this.columns = config.config.columns;
     }
 
     this.entity = config.entity;
-    this.records = (
-      await this.entityMapper.loadType<RecurringActivity>(RecurringActivity)
-    ).filter((activity) => activity.linkedGroups.includes(this.entity.getId()));
+    this.initializeRelatedActivities();
     this.entityMapper
       .receiveUpdates(RecurringActivity)
       .subscribe((updateEntity) => {
         if (updateEntity.type === "update") {
-          this.records = this.records.filter((activity) =>
-            activity.linkedGroups.includes(this.entity.getId())
-          );
+          this.initializeRelatedActivities();
```

Comment:
seems a bit of an overkill to reload every entity of that type - we might be able to just replace (i.e. update) the entity that was updated? Could we use the same as in "EducationalMaterials" for example: `this.records = applyUpdate(this.records, update);`?

But otherwise I think the approach here is an acceptable tradeoff.